### DISCUSSION TITLE
2023-35: 从飞书导出日历到 Fastmail

### DISCUSSION THREAD

**Discussion by giscus** [Xuanwo/blog#410] [Announcements]
# 2023-35: 从飞书导出日历到 Fastmail

An infrastructure engineer, focused on distributed storage system

https://xuanwo.io/reports/2023-35/

**Comment by ImSingee** [comment]
那么问题来了🌚Windows/MacOS 是怎么兼容这么有问题的实现的🌚

**Reply by Xuanwo** [reply]
我觉得应该是反过来：这服务实现的时候应该就是对着 macOS/Windows 实现的

**Comment by hillsdong** [comment]
哈哈，我在用golang读飞书的caldav，感谢分享

**Comment by jason5ng32** [comment]
分享一下我的方法：

1. 在服务器上安装 vdirsyncer ，这个工具可以同步 CalDAV 的内容，在同步设置里，不需要先找到 UUID，可以直接用飞书提供的 URL。
2. 写一个 Python 脚本，将 vdirsyncer 同步的内容合并成单一的 ics 文件。
3. 将 ics 文件放到一个地址稍微复杂一点的 http 目录里，可以外部访问。
4. 写一个 run.sh 脚本，通过 crontab 每 10 分钟执行一次 vdirsyncer 同步和日历文件合成。
5. 在 Google 日历或本地直接订阅自己的 ics 地址。

**Reply by nemofq** [reply]
这个思路靠谱！

Google 日历订阅 ics 的同步频率不太可控（越热门的似乎越可靠），可以用这个项目 https://github.com/derekantrican/GAS-ICS-Sync 在 Google App Script 上跑个定时脚本来同步 ics 到 Google 日历。

**Comment by nemofq** [comment]
参考 Jason Ng 的思路，写了个开源小工具可以定时同步 CalDAV 到 ics 文件，支持部署到 Vercel：
https://github.com/nemofq/caldav-to-ics/
